components/espcoredump/src/core_dump_binary.c

```diff
@@ -185,6 +185,11 @@ static esp_err_t esp_core_dump_write_binary(void)
     hdr.version   = COREDUMP_VERSION_BIN_CURRENT;
     hdr.tcb_sz    = tcb_sz;
     hdr.chip_rev  = efuse_hal_chip_revision();
+    dump_hdr.boot_time = esp_timer_get_time();
+
+    struct timeval tv;
+    gettimeofday(&tv, NULL);
```

Comment:
I'm not sure it is safe to call `gettimeofday` from the panic handler. Internally, `gettimeofday` may attempt to acquire a mutex (e.g. [here](https://github.com/espressif/esp-idf/blob/d4cd437ede613fffacc06ac6d6c93a083829022f/components/newlib/time.c#L92-L94) and a couple other places), which in general might not be possible in the panic handler. (Panic handler might be invoked from a high-priority ISR, such as an interrupt WDT ISR).

Reply:
Perhaps the best alternative I can suggest is to handle this at the application level.

You can designate a static variable with `COREDUMP_DRAM_ATTR` to make sure it gets saved in the coredump. Then, set up a periodic timer to save the `gettimeofday` output into that variable. If a crash happens, you will get the latest value of that variable inside the dump.